src/site/markdown/server/manage-guice-distributed-james.md

```diff
@@ -260,3 +261,70 @@ by setting the parameter `elasticsearch.index.mailbox.name` to the name of your
 re-creates index upon restart
 
 _Note_: keep in mind that reindexing can be a very long operation depending on the volume of mails you have stored.
+
+## Solving cassandra inconsistencies
+
+By choosing the cassandra, it means denormalization is the common modeling pattern to be using. 
+You may see some data is duplicated among tables. Especially, with tables having close relationships. 
+Some operations require multiple table writes, if there is any failure upon in the middle of an operation. 
+It could bring an inconsistent state to the data. The consequence could be dirty reads, unexpected failing writes.
+
+Because of the lack of transactions, it's hard to prevent these kind of issues. We had developed some features to 
+fix some existing cassandra inconsistency issues that had been reported to James. 
+
+There are typical type inconsistencies:
+ - [RRT (RecipientRewriteTable) mapping sources](#rrt-recipientrewritetable-mapping-sources)
+ - [Jmap message fast view projections](#jmap-message-fast-view-projections)
+ - [Mailboxes](#mailboxes)
+
+### RRT (RecipientRewriteTable) mapping sources
+
+`rrt` and `mappings_sources` tables store information about address mappings. 
+While `rrt` is the source of truth and `mappings_sources` is the projection table containing all 
+mapping sources.
+
+#### How to detect the inconsistencies
+
+Right now there's no tool for detecting that, we're proposing a development plan on that.  
+By the mean time, the recommendation is to execute the `SolveInconsistencies` task below 
+in a regular basis. 
+
+#### How to solve
+
+Execute the Cassandra mapping `SolveInconsistencies` task described in [webadmin documentation](https://james.apache.org/server/manage-webadmin.html#Operations_on_mappings_sources) 
+
+### Jmap message fast view projections
+
+When you read a Jmap message, some calculated properties are expected to be fast to retrieve, like `preview`, `hasAttachment`. 
+James does it by pre-calculating and storing them into a message projection table(`message_fast_view_projection`). 
```

Comment:
does what?

Reply:
does that `fast retrieving capability` I should be more clear about it

Reply:
It's actually not grammatically correct AFAICT. you can replace does by achieves and it should work